deploy/tests/conftest.py

```diff
@@ -83,3 +86,14 @@ def patch(self, task_name: str, wrap_task=False, wrap_config=False) -> firesim.T
 
     return TaskMocker(mocker)
 
+
+#References: https://stackoverflow.com/questions/44718670/mock-s3-decorating-pytest-fixture
+@pytest.fixture()
+def mock_s3_client():
+    # setup: start moto server
+    mocks3 = mock_s3()
+    mocks3.start()
+    s3_client = boto3.client('s3')
+    yield s3_client
+    # teardown: stop moto server
+    mocks3.stop()
```

Comment:
Do you need to use `@pytest.yield_fixture` instead of the plain `pytest.fixture`?  I don't know the difference but I've also never written a fixture that yielded.

I have a preference for using `mock_s3` as a context manager as in https://github.com/spulec/moto/issues/620#issuecomment-613454360

I see how you need the fixture like this if you want to build up other fixtures that setup your testing state (i.e. create the bucket and 'upload' test collateral into it)

Reply:
Not necessarily, a context manager will do just fine. I just wanted to add this fixture because it seemed like something that could be useful for other tests in the future.

But we should probably add it when it becomes necessary so we don't keep it around while it's unused.